nano/node/bootstrap/bootstrap_server.cpp

```diff
@@ -9,31 +10,53 @@
 #include <nano/store/component.hpp>
 #include <nano/store/confirmation_height.hpp>
 
-// TODO: Make threads configurable
-nano::bootstrap_server::bootstrap_server (nano::store::component & store_a, nano::ledger & ledger_a, nano::network_constants const & network_constants_a, nano::stats & stats_a) :
+nano::bootstrap_server::bootstrap_server (bootstrap_server_config const & config_a, nano::store::component & store_a, nano::ledger & ledger_a, nano::network_constants const & network_constants_a, nano::stats & stats_a) :
+	config{ config_a },
 	store{ store_a },
 	ledger{ ledger_a },
 	network_constants{ network_constants_a },
-	stats{ stats_a },
-	request_queue{ stats, nano::stat::type::bootstrap_server, nano::thread_role::name::bootstrap_server, /* threads */ 1, /* max size */ 1024 * 16, /* max batch */ 128 }
+	stats{ stats_a }
 {
-	request_queue.process_batch = [this] (auto & batch) {
-		process_batch (batch);
+	queue.max_size_query = [this] (auto const & origin) {
+		return config.max_queue;
+	};
+
+	queue.priority_query = [this] (auto const & origin) {
+		return size_t{ 1 };
 	};
 }
 
 nano::bootstrap_server::~bootstrap_server ()
 {
+	debug_assert (threads.empty ());
 }
 
 void nano::bootstrap_server::start ()
 {
-	request_queue.start ();
+	debug_assert (threads.empty ());
+
+	for (auto i = 0u; i < config.threads; ++i)
+	{
+		threads.push_back (std::thread ([this] () {
+			nano::thread_role::set (nano::thread_role::name::bootstrap_server);
+			run ();
+		}));
+	}
 }
 
 void nano::bootstrap_server::stop ()
 {
```

Comment:
Might be good to have an assert here to check that it is not already stopped.

Reply:
In contrast to start, calling stop multiple times is correct and should behave well.